MudNavMenu: Horizontal alignment

Anu6is opened the pull request:
## Description
Adds an option to align the nav menu horizontally.
- Aligns `MudNavMenu` horizontally 
- Nested `MudNavGroups` use a popover to display nested items. 
- `Bordered` option adds a border to the **bottom** instead of the **end**

Resolves #474 

## How Has This Been Tested?
Visually

## Type of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

![image](https://github.com/user-attachments/assets/30b53f43-cddc-469e-8686-063a6cf02eb2)

## Checklist
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

Anu6is committed:
MudNavMenu: Horizontal alignment

danielchalmers reviewed (commented):
Haven't tested it fully but looks really good! Do you think you could replace the top bar in the docs with it so we can dogfood it?

![image](https://github.com/user-attachments/assets/e9d9b996-e248-4352-8828-d38a4fefd0fe)

When is `ParentNavMenu` useful? Could you show an example?

Anu6is commented:
ParentNavMenu is used for the nested MudNavGroups. It ensures that each group has its own popover.

If I have some time today I'll check how well it can substitute for the docs menu.

Anu6is committed:
Merge branch 'MudBlazor:dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

danielchalmers commented:
Why do we want nested nav menus instead of utilizing MudMenu?
If that's removed we should update the example to use an icon button instead and make sure the app bar flows correctly with mixed components.

I'm a little concerned about cannibalizing other components.

~~It seems like the title should be moved to the app bar and the app bar should contain two horizontal menus with the spacer being also moved to the app bar. But I think that's just a user choice like including them in the vertical one.~~

---

Items without icons have space reserved:

![image](https://github.com/user-attachments/assets/bb62446b-8362-4caa-a015-f11890546c01)

Does not flow on narrow screens:

![image](https://github.com/user-attachments/assets/2b2adce6-1f54-4c4d-8a41-d7d46c31a856)

Anu6is commented:
The nested menu was just to show that the [Sub Groups](https://mudblazor.com/components/navmenu#sub-groups) example still work horizontally. 

Will look into the other notes

danielchalmers commented:
Is it possible to lose ParentNavMenu and use a MudMenu internally instead?

Anu6is commented:
Would that change be applied to the regular vertical menu as well. My intention was to keep both the same while just changing the orientation.

danielchalmers commented:
> Would that change be applied to the regular vertical menu as well. My intention was to keep both the same while just changing the orientation.

Vertical should stay the same but the horizontal submenus could use a MudMenu internally instead of the custom implementation, if I'm understanding right. Would like to avoid complication with the ParentNavMenu cascading parameter if possible

But I guess how I can see this will get complicated without #10255. Will have to think about it more

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
review tweaks

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is commented:
@danielchalmers actually won't be able to use a Menu, the actual purpose of [that section](https://github.com/MudBlazor/MudBlazor/pull/10222/files#diff-02eb8e0100a32b76fbd899994d3bcd6b547f662b1add5ef65030fcac81f535e3R46) code is the `NavGroup` is to create new `NavMenu` using the `NavLinks` provided. There's no way to map the nav links to menu items. The custom popover stuff was added because using collapse works for the vertical menu but not horizontal. 

Adjust the example to a more practical one to show this. Removed the nested groups and did a single nav group. 

Also fixed the no icon padding.

Anu6is committed:
Update NavMenuHorizontalExample.razor

danielchalmers commented:
Me and @Garderoben are concerned about this cannibalizing other components, Ex: the title and spacer should be in the app bar and the gear submenu should be a MudMenu. The NavMenu is being contorted into something it wasn't designed for.

Correct me if I'm wrong but it seems like if the nav link buttons were able to be used standalone as shown in the screenshot, you could accomplish the same demo with much less code and more appropriate controls - This could be done by applying different CSS based on whether it's in a vertical nav menu or it's on its own. What do you think? Would that work?

Anu6is commented:
I didn't update the screenshot when I adjusted the [example](https://github.com/MudBlazor/MudBlazor/pull/10222/files#diff-0749af4065258e1b8b346ac9b974c45c552c639c2cf3e21dc1a29eda55e2a9e7R6), however, the title and spacer are now in the app bar and not the nav menu. The majority of the changes are indeed css. The additional code change was to avoid [this](https://github.com/MudBlazor/MudBlazor/issues/474#issuecomment-754518646) problem pointed out in the original issue.

![image](https://github.com/user-attachments/assets/2e21fa76-8698-41c2-8c78-7d5b2e104739)

danielchalmers commented:
MudMenu has FullWidth now and I've got more goodies on the way #10452 

Your links look great:

![image](https://github.com/user-attachments/assets/3dc89843-0f8d-4bce-b860-cddbce738b6b)

My point was that could be the default style unless they're placed in a NavMenu, which then they would turn into what we have now. Then no changes are needed to the nav menu itself and you don't have to use it to get the buttons

Anu6is commented:
ok, understood

Anu6is committed:
Merge branch 'MudBlazor:dev' into Horizontal-NavMenu

Anu6is committed:
rework

Anu6is committed:
Merge branch `MudBlazor:dev` into Horizontal-NavMenu

Anu6is committed:
cleanup css after dev merge

Anu6is committed:
remove duplication

Anu6is commented:
@danielchalmers I know we spoke about styling the `MudNavLinks` themselves so users could just add those to the app bar and the default renders like my initial mock up. However, while going down that path I realized all the styling properties, are on the `MudNavMenu` so users wouldn't actually be able to easily modify things like Color, Dense, Margin etc.

So while I did remove the previous cascading values complexities, this still maintains nesting the `MudNavLinks` in a `MudNavMenu`

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

danielchalmers commented:
Sorry for the delay, I'm not sure when I'm going to be able to get to it but I have it saved

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

Anu6is committed:
Merge branch 'dev' into Horizontal-NavMenu

danielchalmers commented:
I think we can't afford to add an option like `Horizontal` because it encourages cannibalizing other components like mentioned above. We would want a type of styled link button, maybe using MudButton directly to base the styles on and bypass the nav stuff (not sure). We just ultimately can't add a new horizontal layout paradigm for a small feature because users will want to expand it in future PRs and we're back to the original problem.
